### PR TITLE
ZH-trans: add guestbook.md

### PR DESCRIPTION
**Pull request by xichengliudui**
>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> Please delete this note before submitting the pull request.
>
> For 1.14 Features: set Milestone to 1.14 and Base Branch to dev-1.14
> 
> For Chinese localization, base branch to release-1.12
>
> For Korean Localization: set Base Branch to dev-1.13-ko.<latest team milestone>
>
> Help editing and submitting pull requests:
> https://kubernetes.io/docs/contribute/start/#improve-existing-content.
>
> Help choosing which branch to use:
> https://kubernetes.io/docs/contribute/start#choose-which-git-branch-to-use.
>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>

### CODE REVIEW

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 1]
```diff
@@ -0,0 +1,621 @@
...
+本教程向您展示如何使用 Kubernetes 和 [Docker](https://www.docker.com/) 构建和部署
+一个简单的多层 web 应用程序。本例由以下组件组成：
+
+<!--
+* A single-instance [Redis](https://redis.io/) master to store guestbook entries 
+* Multiple [replicated Redis](https://redis.io/topics/replication) instances to serve reads 
+* Multiple web frontend instances
+-->
+
+* 一个单实例 [Redis](https://redis.io/) 主程序来存储留言板条目
```
Comment (tengqm) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]: master 在这里是指 redis 的主节点，不是主程序

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 2]
```diff
@@ -0,0 +1,621 @@
...
+
+<!--
+* Start up a Redis master.
+* Start up Redis slaves.
+* Start up the guestbook frontend.
+* Expose and view the Frontend Service.
+* Clean up.
+-->
+
+* 启动 Redis 主服务器。
```
Comment (tengqm) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]: 主节点

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 3]
```diff
@@ -0,0 +1,621 @@
...
+* 启动 Redis slaves。
```
Comment (tengqm) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]: 启动  Redis 工作节点

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 4]
```diff
@@ -0,0 +1,621 @@
...
+     ```shell
+     kubectl logs -f POD-NAME
+     ```
+
+{{< note >}}
+
+<!--
+Replace POD-NAME with the name of your Pod.
+-->
+将您的 Pod 名称替换为 POD-NAME。
```
Comment (tengqm) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]: 顺序反了

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 5]
```diff
@@ -0,0 +1,621 @@
...
+-->
+这个清单文件创建了一个名为 `Redis-master` 的服务，其中包含一组与前面定义的标签匹配的标签，因此服务将网络流量路由到 Redis 主 Pod。
+
+{{< /note >}}
+
+<!--
+## Start up the Redis Slaves
+-->
+
+## 启动 Redis Slaves
```
Comment (tengqm) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]: slaves 这个词很忌讳的，可以考虑翻译为工作节点

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 6]
```diff
@@ -0,0 +1,621 @@
...
+<!--
+### Creating the Redis Slave Deployment
+-->
+
+### 创建 Redis Slave Deployment
+
+<!--
+Deployments scale based off of the configurations set in the manifest file. In this case, the Deployment object specifies two replicas. 
+-->
+Deployments 根据清单文件中设置的配置进行伸缩。在这种情况下，部署对象指定两个副本。
```
Comment (tengqm) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]: Deployment 不要翻译

File: content/zh/docs/tutorials/stateless-application/guestbook.md (hunk 5 above)
Comment (xichengliudui) [reply content/zh/docs/tutorials/stateless-application/guestbook.md]: 赞成，翻译时思考了 `slaves` 很久。

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 7]
```diff
@@ -0,0 +1,621 @@
...
+本教程向您展示如何使用 Kubernetes 和 [Docker](https://www.docker.com/) 构建和部署
+一个简单的多层 web 应用程序。本例由以下组件组成：
+
+<!--
+* A single-instance [Redis](https://redis.io/) master to store guestbook entries 
+* Multiple [replicated Redis](https://redis.io/topics/replication) instances to serve reads 
+* Multiple web frontend instances
+-->
+
+* 一个单实例 [Redis](https://redis.io/) 主节点来存储留言板条目
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
* 单实例 [Redis](https://redis.io/) 主节点保存留言板条目
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 8]
```diff
@@ -0,0 +1,621 @@
...
+* 多个[复制的 Redis](https://redis.io/topics/replication) 实例用于提供读取
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
* 多个[复制 Redis](https://redis.io/topics/replication) 节点用来读取数据
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 9]
```diff
@@ -0,0 +1,621 @@
...
+<!--
+* Start up a Redis master.
+* Start up Redis slaves.
+* Start up the guestbook frontend.
+* Expose and view the Frontend Service.
+* Clean up.
+-->
+
+* 启动 Redis 主节点。
+* 启动 Redis 工作节点。
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
* 启动 Redis 从节点。
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 10]
```diff
@@ -0,0 +1,621 @@
...
+
+{{% /capture %}}
+
+{{% capture lessoncontent %}}
+
+<!--
+## Start up the Redis Master
+-->
+
+## 启动主 Redis
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
## 启动 Redis 主节点
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 11]
```diff
@@ -0,0 +1,621 @@
...
+<!--
+The guestbook application uses Redis to store its data. It writes its data to a Redis master instance and reads data from multiple Redis slave instances.
+-->
+留言板应用程序使用 Redis 存储数据。它将数据写入一个 Redis 主实例，并从多个 Redis 读取数据。
+
+<!--
+### Creating the Redis Master Deployment
+-->
+
+### 创建 Redis 主 Deployment
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
### 创建 Redis 主节点的 Deployment
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 12]
```diff
@@ -0,0 +1,621 @@
...
+
+<!--
+The manifest file, included below, specifies a Deployment controller that runs a single replica Redis master Pod.
+-->
+下面包含的清单文件指定了一个 Deployment 控制器，该控制器运行一个 Redis 主 Pod 副本。
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
下面包含的清单文件指定了一个 Deployment 控制器，该控制器运行一个 Redis 主节点 Pod 副本。
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 13]
```diff
@@ -0,0 +1,621 @@
...
+1. 从 `redis-master-deployment.yaml` 文件中应用 Redis 主 Deployment：
+
+      ```shell
+      kubectl apply -f https://k8s.io/examples/application/guestbook/redis-master-deployment.yaml
+      ```
+
+<!--
+1. Query the list of Pods to verify that the Redis Master Pod is running:
+-->
+1. 查询 Pod 列表以验证 Redis 主 Pod 是否正在运行：
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
1. 查询 Pod 列表以验证 Redis 主节点 Pod 是否正在运行：
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 14]
```diff
@@ -0,0 +1,621 @@
...
+
+      ```shell
+      NAME                            READY     STATUS    RESTARTS   AGE
+      redis-master-1068406935-3lswp   1/1       Running   0          28s
+      ```
+
+<!--
+1. Run the following command to view the logs from the Redis Master Pod:
+-->
+1. 运行以下命令查看 Redis 主 Pod 中的日志：
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
1. 运行以下命令查看 Redis 主节点 Pod 中的日志：
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 15]
```diff
@@ -0,0 +1,621 @@
...
+-->
+将 POD-NAME 替换为您的 Pod 名称。
+
+{{< /note >}}
+
+<!--
+### Creating the Redis Master Service
+-->
+
+### 创建 Redis 主服务
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
### 创建 Redis 主节点的服务
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 16]
```diff
@@ -0,0 +1,621 @@
...
+
+<!--
+The guestbook applications needs to communicate to the Redis master to write its data. You need to apply a [Service](/docs/concepts/services-networking/service/) to proxy the traffic to the Redis master Pod. A Service defines a policy to access the Pods.
+-->
+留言板应用程序需要与 Redis 主机通信以写入其数据。您需要应用服务来代理 Redis 主 Pod 的流量。服务定义了访问 Pod 的策略。
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
留言板应用程序需要往 Redis 主节点中写数据。因此，需要创建 [Service](/docs/concepts/services-networking/service/) 来代理 Redis 主节点 Pod 的流量。Service 定义了访问 Pod 的策略。
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 17]
```diff
@@ -0,0 +1,621 @@
...
+
+{{< codenew file="application/guestbook/redis-master-service.yaml" >}}
+
+<!--
+1. Apply the Redis Master Service from the following `redis-master-service.yaml` file: 
+-->
+1. 从以下 `redis-master-service.yaml` 文件应用 Redis 主服务：
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
1. 使用下面的 `redis-master-service.yaml` 文件创建 Redis 主节点的服务：
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 18]
```diff
@@ -0,0 +1,621 @@
...
+
+      ```shell
+      kubectl apply -f https://k8s.io/examples/application/guestbook/redis-master-service.yaml
+      ```
+
+<!--
+1. Query the list of Services to verify that the Redis Master Service is running:
+-->
+1. 查询服务列表验证 Redis 主服务是否正在运行：
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
1. 查询服务列表验证 Redis 主节点服务是否正在运行：
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 19]
```diff
@@ -0,0 +1,621 @@
...
+      kubernetes     ClusterIP   10.0.0.1     <none>        443/TCP    1m
+      redis-master   ClusterIP   10.0.0.151   <none>        6379/TCP   8s
+      ```
+
+{{< note >}}
+
+<!--
+This manifest file creates a Service named `redis-master` with a set of labels that match the labels previously defined, so the Service routes network traffic to the Redis master Pod.   
+-->
+这个清单文件创建了一个名为 `Redis-master` 的服务，其中包含一组与前面定义的标签匹配的标签，因此服务将网络流量路由到 Redis 主 Pod。
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
这个清单文件创建了一个名为 `Redis-master` 的 Service，其中包含一组与前面定义的标签匹配的标签，因此服务将网络流量路由到 Redis 主节点 Pod 上。
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 20]
```diff
@@ -0,0 +1,621 @@
...
+
+{{< /note >}}
+
+<!--
+## Start up the Redis Slaves
+-->
+
+## 启动 Redis 工作节点
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
## 启动 Redis 从节点
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 21]
```diff
@@ -0,0 +1,621 @@
...
+
+<!--
+Although the Redis master is a single pod, you can make it highly available to meet traffic demands by adding replica Redis slaves.
+-->
+尽管主 Redis 是一个单独的 pod，但是您可以通过添加副本 Redis slave 来使其高度可用，以满足流量需求。
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
尽管 Redis 主节点是一个单独的 pod，但是您可以通过添加 Redis 从节点的方式来使其高可用性，以满足流量需求。
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 22]
```diff
@@ -0,0 +1,621 @@
...
+
+<!--
+### Creating the Redis Slave Deployment
+-->
+
+### 创建 Redis Slave Deployment
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
### 创建 Redis 从节点 Deployment
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 23]
```diff
@@ -0,0 +1,621 @@
...
+      redis-master-1068406935-3lswp   1/1       Running             0          1m
+      redis-slave-2005841000-fpvqc    0/1       ContainerCreating   0          6s
+      redis-slave-2005841000-phfv9    0/1       ContainerCreating   0          6s
+      ```
+
+<!--
+### Creating the Redis Slave Service
+-->
+
+### 创建 Redis Slave 服务
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
### 创建 Redis 从节点的 Service
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 24]
```diff
@@ -0,0 +1,621 @@
...
+
+<!--
+The guestbook application needs to communicate to Redis slaves to read data. To make the Redis slaves discoverable, you need to set up a Service. A Service provides transparent load balancing to a set of Pods.
+-->
+留言板应用程序需要与 Redis 从服务器通信以读取数据。要使 Redis 从服务器能够被发现，
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
留言板应用程序需要从 Redis 从节点中读取数据。
为了便于 Redis 从节点可发现，
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 25]
```diff
@@ -0,0 +1,621 @@
...
+您需要设置一个服务。服务为一组 Pod 提供透明的负载均衡。
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
您需要设置一个 Service。Service 为一组 Pod 提供负载均衡。
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 26]
```diff
@@ -0,0 +1,621 @@
...
+1. 从 `frontend-deployment.yaml` 应用前端 Deployment 文件：
+
+      ```shell
+      kubectl apply -f https://k8s.io/examples/application/guestbook/frontend-deployment.yaml
+      ```
+
+<!--
+1. Query the list of Pods to verify that the three frontend replicas are running:
+-->
+1. 查询 Pod 列表 验证三个前端副本是否正在运行：
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
1. 查询 Pod 列表，验证三个前端副本是否正在运行：
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 27]
```diff
@@ -0,0 +1,621 @@
...
+If you want guests to be able to access your guestbook, you must configure the frontend Service to be externally visible, so a client can request the Service from outside the container cluster. Minikube can only expose Services through `NodePort`.  
+-->
+如果您希望客人能够访问您的留言板，您必须将前端服务配置为外部可见的，以便客户机可以从容器集群之外请求服务。Minikube 只能通过 `NodePort` 公开服务。
+
+{{< note >}}
+
+<!--
+Some cloud providers, like Google Compute Engine or Google Kubernetes Engine, support external load balancers. If your cloud provider supports load balancers and you want to use it, simply delete or comment out `type: NodePort`, and uncomment `type: LoadBalancer`. 
+-->
+一些云提供商，如谷歌计算引擎或谷歌 Kubernetes 引擎，支持外部负载均衡器。如果您的云提供商支持负载均衡器，并且您希望使用它，
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
一些云提供商，如 Google Compute Engine 或 Google Kubernetes Engine，支持外部负载均衡器。如果您的云提供商支持负载均衡器，并且您希望使用它，
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 28]
```diff
@@ -0,0 +1,621 @@
...
+只需删除或注释掉 `type: NodePort`，并取消注释 `type: LoadBalancer`。
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
只需删除或注释掉 `type: NodePort`，并取消注释 `type: LoadBalancer` 即可。
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 29]
```diff
@@ -0,0 +1,621 @@
...
+<!--
+## Scale the Web Frontend 
+-->
+
+## 扩展 Web 前端
+
+<!--
+Scaling up or down is easy because your servers are defined as a Service that uses a Deployment controller.
+-->
+向上或向下伸缩很容易，因为您的服务器被定义为使用 Deployment 控制器的服务。
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
伸缩很容易是因为服务器本身被定义为使用一个 Deployment 控制器的 Service。
```

File: content/zh/docs/tutorials/stateless-application/guestbook.md [hunk 30]
```diff
@@ -0,0 +1,621 @@
...
+
+<!--
+* Complete the [Kubernetes Basics](/docs/tutorials/kubernetes-basics/) Interactive Tutorials
+* Use Kubernetes to create a blog using [Persistent Volumes for MySQL and Wordpress](/docs/tutorials/stateful-application/mysql-wordpress-persistent-volume/#visit-your-new-wordpress-blog) 
+* Read more about [connecting applications](/docs/concepts/services-networking/connect-applications-service/)
+* Read more about [Managing Resources](/docs/concepts/cluster-administration/manage-deployment/#using-labels-effectively)
+-->
+
+* 完成 [Kubernetes Basics](/docs/tutorials/kubernetes-basics/) 交互式教程
+* 使用 Kubernetes 创建一个博客，使用[ MySQL 和 Wordpress 的持久卷](/docs/tutorials/stateful-application/mysql-wordpress-persistent-volume/#visit-your-new-wordpress-blog)
```
Comment (chenrui333) [inline content/zh/docs/tutorials/stateless-application/guestbook.md]:
```suggestion
* 使用 Kubernetes 创建一个博客，使用 [MySQL 和 Wordpress 的持久卷](/docs/tutorials/stateful-application/mysql-wordpress-persistent-volume/#visit-your-new-wordpress-blog)
```

**Comment by xichengliudui** [comment]
已更新。

**Comment by chenrui333** [comment]
cc @tengqm

**Pull request merged**